### ISSUE TITLE
request error Context conflict

### ISSUE THREAD

**Issue by uktamka** [nuxt/nuxt#20070]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.17.1`
- Nuxt Version:     `3.3.3`
- Nitro Version:    `2.3.2`
- Package Manager:  `pnpm@7.28.0`
- Builder:          `vite`
- User Config:      `srcDir`, `debug`, `typescript`, `app`, `experimental`, `vite`, `modules`, `i18n`, `tailwindcss`, `vueuse`, `routeRules`, `imports`, `components`, `runtimeConfig`
- Runtime Modules:  `@nuxt/devtools@0.1.6`, `@nuxtjs/i18n@8.0.0-beta.9`, `@nuxtjs/tailwindcss@6.6.5`, `@vueuse/nuxt@9.13.0`, `@nuxt/image-edge@1.0.0-27919678.2f5b64b`, `@pinia/nuxt@0.4.7`
- Build Modules:    `-`
------------------------------

### Reproduction

Has not

### Describe the bug

[nuxt] [request error] [unhandled] [500] Context conflict
  at checkConflict (./server/node_modules/unctx/dist/index.mjs:6:13)
  at Object.call (./server/node_modules/unctx/dist/index.mjs:31:7)
  at contextCaller (./server/chunks/app/_nuxt/entry-a7e1e255.mjs:333:26)
  at Hookable.callHookWith (./server/node_modules/hookable/dist/index.mjs:198:20)
  at Hookable.nuxtApp.hooks.callHook (./server/chunks/app/_nuxt/entry-a7e1e255.mjs:336:63)
  at ./server/chunks/handlers/renderer.mjs:372:32
  at processTicksAndRejections (node:internal/process/task_queues:96:5)
  at async ./server/chunks/handlers/renderer.mjs:16:22
  at async Object.handler (./server/node_modules/h3/dist/index.mjs:1247:19)
  at async Server.toNodeHandle (./server/node_modules/h3/dist/index.mjs:1322:7)

**Comment by danielroe** [comment]
Would you provide a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by SVV-team** [comment]
<img width="945" alt="Снимок экрана 2023-04-09 в 20 55 47" src="https://user-images.githubusercontent.com/48009651/230788714-e4415556-b996-4f11-b434-909c40bdbd2d.png">
Same problem in PM2 logs

**Comment by dokaNc** [comment]
I have the same problem
I'm in DEV not PROD

<img width="695" alt="Capture d’écran 2023-04-10 à 19 09 26" src="https://user-images.githubusercontent.com/47001214/230953483-ed84e3a1-5f92-4725-9dc0-b35df3a286c6.png">

**Comment by aymericingargiola** [comment]
I have the same issue, for me it seems to happen when i load a page (first load) where pinia store is called

**Comment by EduardoMateos** [comment]
I can't reproduce it in my development environment. I don't know what causes it. I only see it in the production logs. any ideas?

**Comment by qiuyoo** [comment]
Return to normal after refresh，I have this problem here too

**Comment by EduardoMateos** [comment]
@danielroe  https://stackblitz.com/edit/nuxt-issue-20070

It is also related to the package "@nuxtjs/i18n" and "useCookie".

**1**. On a clean nuxt 3 project, install:
"@nuxtjs/i18n": "^8.0.0-beta.10"

**2**. Create global middleware like this:
```
export default defineNuxtRouteMiddleware(() => {
  const session = useCookie<string>("uuidSession");
});
```
**3**. Use siege and gtimeout to send multiple concurrent requests:
brew install siege                            
brew install coreutils
gtimeout 30 siege -c 5 http://localhost:3000


then sometimes I will get the error:
[nuxt] [request error] [unhandled] [500] Context conflict

**Comment by EduardoMateos** [comment]
Open issue in i18n package: https://github.com/nuxt-modules/i18n/issues/1984

**Comment by c-schwan** [comment]
@danielroe digging a little bit deeper into the problem i figure out that it throws an Error on unctx while checking the instance vs current context. I belive its not an i18n error.

**Comment by danielroe** [comment]
@c-schwan It's correct that an error is thrown if a Vue or Nuxt composable is called in the wrong place. See https://nuxt.com/docs/guide/concepts/auto-imports#using-vue-and-nuxt-composables.

**Comment by EduardoMateos** [comment]
> nuxt-modules/i18n#1984

I have tried as @c-schwan  told me and it works perfectly in version 3.3.0. 

![Captura de pantalla 2023-04-13 a las 7 26 09](https://user-images.githubusercontent.com/11529050/231662284-3becc85c-bf3d-4dda-b69a-20adee893e61.png)

**Comment by EduardoMateos** [comment]
I specify a little more:

3.3.0 ->works
3.3.1->works
3.3.2-> error
3.4.0->error

**Comment by c-schwan** [comment]
> @c-schwan It's correct that an error is thrown if a Vue or Nuxt composable is called in the wrong place. See https://nuxt.com/docs/guide/concepts/auto-imports#using-vue-and-nuxt-composables.

@danielroe its strange because it happened randomly (multiple concurrent requests) 3/10 throws an context error, the normal error "Nuxt instance is unavailable" it not thrown.

**Comment by danielroe** [comment]
Indeed, I think this might be an issue we need to investigate.

cc: @pi0

**Comment by EduardoMateos** [comment]
any update? :) 

thanks guys

**Comment by SebastianBoergers** [comment]
> I specify a little more:
> 
> 3.3.0 ->works 3.3.1->works 3.3.2-> error 3.4.0->error

FYI: We already had those in 3.3.1.

I'm not super sure if this is related to this problem specifically, since we have another construction site with Nuxt Instance Unavailable, but what happens with our App is quite...odd:

So imagine you open the website and just take 5 links and open them in a new Tab at the same time.

What happens so far was, that in our App.vue everything seems fine and all URLs/Paths look right.  
But down the road in an (async) composable it's suddenly all the same URL/Path.

It'll look something like this:
```
app-setup -> /a
app-setup -> /b
app-setup -> /c
composable -> /a
composable -> /a
composable -> /a
```

This results either complete Errors with the Context-Issue OR you get a fully rendered page but with the content of `/a` even though you're clearly on `/b` (in your browser-url).

Now reproducing this...well. I'm still looking into where things actually go south and what's related to what, but maybe this helps you guys a little?

**Comment by SebastianBoergers** [comment]
So, for us it **seem's** our own Plugin that does some stuff with the router due to our site being used on multiple domains + languages, but our url structure won't indicate that.

```
abc.net/home -> pages/abc/home
def.net/home -> pages/def/home
```
And then since we use i18n but we DONT want prefixed URLs, we remove the prefixes from the routes in our Plugin.

Basically we're getting our routes, remove a lot of them, and only give back the right ones for the domain/language combination, that's the gist of it.

What's funny is, that _useRoute().path_ will provide the right path, while _useRouter().router.currentRoute.value.path_ will have a wrong path from a previous request.

Sorry for the bad explanation, we're in the middle of troubleshooting this.

**Comment by burda-osowiecki** [comment]
I have similar or maybe even the same issue.
I made a reproduction ;)
https://github.com/nuxt/nuxt/issues/20473

**Comment by ilyaDegtyarenko** [comment]
+

**Comment by SimonXunming** [comment]
@EduardoMateos Hi , Have you solved your problem? We've also been dealing with "[nuxt] [request error] [unhandled] [500] Context conflict" lately. We come up all the time in production and occasionally in development.

**Comment by SimonXunming** [comment]
> I specify a little more:
> 
> 3.3.0 ->works 3.3.1->works 3.3.2-> error 3.4.0->error

@EduardoMateos As you said, the problem was solved after we changed the nuxt version to 3.3.0.
Hope the Nuxt team can solve this problem soon

**Comment by EduardoMateos** [comment]
We are in version 3.3.1 waiting for a fix.
